lib/query.py

```diff
@@ -13,15 +13,20 @@ class Query:
     cond_ops = ['=', '>', '<', 'OP']
     syms = ['SELECT', 'WHERE', 'AND', 'COL', 'TABLE', 'CAPTION', 'PAGE', 'SECTION', 'OP', 'COND', 'QUESTION', 'AGG', 'AGGOPS', 'CONDOPS']
 
-    def __init__(self, sel_index, agg_index, conditions=tuple()):
+    def __init__(self, sel_index, agg_index, conditions=tuple(), ordered=False):
         self.sel_index = sel_index
         self.agg_index = agg_index
         self.conditions = list(conditions)
+        self.ordered = ordered
 
     def __eq__(self, other):
         if isinstance(other, self.__class__):
             indices = self.sel_index == other.sel_index and self.agg_index == other.agg_index
-            conds = [(col, op, cond.lower() if isinstance(cond, str) else cond) for col, op, cond in self.conditions] == [(col, op, cond.lower() if isinstance(cond, str) else cond) for col, op, cond in other.conditions]
+            if other.ordered:
+                conds = [(col, op, str(cond).lower()) for col, op, cond in self.conditions] == [(col, op, str(cond).lower()) for col, op, cond in other.conditions]
```

Comment:
why do we need to stringify all numbers?

Reply:
Last time I check, numbers were always treated as strings during execution, but that’s not the behavior for exact match. It might be possible to look in the table to see which should be considered as numbers and which as strings, but it seems simpler to always convert to string.